### PR TITLE
More detailed defaults for `problem.get_full_vector`

### PR DESCRIPTION
**Pull request by PaulJonasJost**
As the doctoring already explained before, we usually want `problem.x_fixed_vals` as defaults if we update a vector x. I think this should be a default and leaving the nans should be left for gradients (hence another parameter to distinguish the two.

### CODE REVIEW

**Review by Doresic — APPROVED** [review]
Looks good.
Makes more sense for the default to grab problem.x_fixed_vals instead of not using them.

File: pypesto/problem/base.py [hunk 1]
```diff
@@ -343,8 +346,10 @@ def get_full_vector(
             The vector in dimension dim.
         x_fixed_vals: array_like, ndim=1, optional
             The values to be used for the fixed indices. If None, then nans are
-            inserted. Usually, None will be used for grad and
-            problem.x_fixed_vals for x.
+            inserted in case of grad and problem.x_fixed_vals is used in case
+            of x.
+        x_is_grad: bool
+            Whether the vector is a grad or x.
```
Comment (m-philipps) [inline pypesto/problem/base.py:352]:
```suggestion
            If true, x is treated like gradients; if false (default) x is treated as parameters.
```

File: pypesto/problem/base.py [hunk 2]
```diff
@@ -343,8 +346,10 @@ def get_full_vector(
             The vector in dimension dim.
         x_fixed_vals: array_like, ndim=1, optional
             The values to be used for the fixed indices. If None, then nans are
-            inserted. Usually, None will be used for grad and
-            problem.x_fixed_vals for x.
+            inserted in case of grad and problem.x_fixed_vals is used in case
+            of x.
```
Comment (m-philipps) [inline pypesto/problem/base.py:350]:
```suggestion
            The values to be used for the fixed indices. If None and x_is_grad=False, problem.x_fixed_vals is used; for x_is_grad=True, nans are inserted.
```

File: pypesto/problem/base.py [hunk 3]
```diff
@@ -343,8 +346,10 @@ def get_full_vector(
             The vector in dimension dim.
         x_fixed_vals: array_like, ndim=1, optional
             The values to be used for the fixed indices. If None, then nans are
```
Comment (m-philipps) [inline pypesto/problem/base.py:348]:
```suggestion
```

**Review by m-philipps — APPROVED** [review]
Thank you! I left an optional suggestion to extend on the docstring.

File: pypesto/problem/base.py (hunk 1 above)
Comment (PaulJonasJost) [reply pypesto/problem/base.py:352]:
```suggestion
            If true, x is treated as gradients.
```

**Pull request merged**